caddytest/integration/caddyfile_test.go

```diff
@@ -78,14 +78,61 @@ func TestDefaultSNI(t *testing.T) {
   }
   
   127.0.0.1:9443 {
-    tls /caddy.localhost.crt /caddy.localhost.key {
-    }
+    tls /caddy.localhost.crt /caddy.localhost.key
+    respond /version 200 {
+      body "hello from a.caddy.localhost"
+    }	
+  }
+  `, "caddyfile")
+
+	// act and assert
+	caddytest.AssertGetResponse(t, "https://127.0.0.1:9443/version", 200, "hello from a.caddy.localhost")
+}
+
+func xTestDefaultSNIWithNamedHostAndExplicitIP(t *testing.T) {
+
+	// arrange
+	caddytest.InitServer(t, ` 
+  {
+    http_port     9080
+    https_port    9443
+    default_sni   a.caddy.localhost
+  }
+  
+  a.caddy.localhost:9443, 127.0.0.1:9443 {
+    tls /a.caddy.localhost.crt /a.caddy.localhost.key
+    respond /version 200 {
+      body "hello from a.caddy.localhost"
+    }	
+  }
+  `, "caddyfile")
+
+	// act and assert
+	caddytest.AssertGetResponse(t, "https://a.caddy.localhost:9443/version", 200, "hello from a.caddy.localhost")
+	// makes a request with no sni
+	caddytest.AssertGetResponse(t, "https://127.0.0.1:9443/version", 200, "hello from a.caddy.localhost")
+}
+
+func xTestDefaultSNIWithPortMappingOnly(t *testing.T) {
```

Comment:
Intentional to skip this test?

Reply:
I disabled it because it relies on the hostname in the test. What I should do it comment out / delete the request that makes the hostname call instead.
The lack of hostname support is a bit of a pain, I will investigate again once the new CI is up and running.